iam_redpanda_agent.tf

```diff
@@ -0,0 +1,947 @@
+      "arn:aws:iam::${local.aws_account_id}:policy/redpanda-connectors-secrets-manager-*",
+      "arn:aws:iam::${local.aws_account_id}:policy/redpanda-console-secrets-manager-*",
+      "arn:aws:iam::${local.aws_account_id}:policy/redpanda-cloud-storage-manager-*",
+      "arn:aws:iam::aws:policy/Amazon*"
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "iam:GetRole",
+      "iam:PassRole",
+      "iam:ListAttachedRolePolicies",
+      "iam:ListInstanceProfilesForRole",
+      "iam:ListRolePolicies",
+      "iam:ListRoleTags",
+    ]
+    resources = [
+      "arn:aws:iam::${local.aws_account_id}:role/redpanda-cloud-storage-manager-*",
+      "arn:aws:iam::${local.aws_account_id}:role/redpanda-console-secrets-manager-*",
+      "arn:aws:iam::${local.aws_account_id}:role/redpanda-connectors-secrets-manager-*",
+      aws_iam_role.redpanda_agent.arn,
+      aws_iam_role.redpanda_node_group.arn,
+      aws_iam_role.redpanda_utility_node_group.arn,
+      aws_iam_role.connectors_node_group.arn,
+      aws_iam_role.k8s_cluster.arn,
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:*",
+    ]
+    resources = [
+      aws_s3_bucket.management.arn,
+      "${aws_s3_bucket.management.arn}/*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:List*",
+      "s3:Get*",
+    ]
+    resources = [
+      aws_s3_bucket.redpanda_cloud_storage.arn,
+      "${aws_s3_bucket.redpanda_cloud_storage.arn}/*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "dynamodb:DescribeTable",
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:DeleteItem",
+    ]
+    resources = [
+      aws_dynamodb_table.terraform_locks.arn,
+      "${aws_dynamodb_table.terraform_locks.arn}/*",
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "autoscaling:*",
+    ]
+    resources = [
+      "arn:aws:autoscaling:*:${local.aws_account_id}:autoScalingGroup:*:autoScalingGroupName/redpanda*",
+    ]
+    dynamic "condition" {
+      for_each = var.condition_tags
+      content {
+        test     = "StringEquals"
+        variable = "autoscaling:ResourceTag/${condition.key}"
+        values = [
+          condition.value,
+        ]
+      }
+    }
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "ec2:TerminateInstances",
+      "ec2:RebootInstances",
+    ]
+    resources = [
+      # the ID of the instance is not known until after the cluster has been created (and even after that is subject to
+      # change) and does not support user specification of the id or an id prefix
+      "arn:aws:ec2:${var.region}:${local.aws_account_id}:instance/*"
+    ]
+    dynamic "condition" {
+      for_each = var.condition_tags
+      content {
+        test     = "StringEquals"
+        variable = "ec2:ResourceTag/${condition.key}"
+        values = [
+          condition.value,
+        ]
+      }
+    }
+  }
+}
```

Comment:
(Sorry I didn't notice this before) This resource block should not end on line 451, it should end after the next statement block ~line 474